misc/review/MidiMessage32.idl

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+// ============================================================================
+// This is part of the Windows MIDI Services App API and should be used
+// in your Windows application via an official binary distribution.
+// Further information: https://github.com/microsoft/MIDI/
+// ============================================================================
+
+#include "namespace_defines.h"
+#include "midl_defines.h"
+MIDI_IDL_IMPORT
+
+import "IMidiUniversalPacket.idl";
+
+namespace MIDI_MIDL_NAMESPACE
+{
+    MIDI_API_CONTRACT_v1
+    unsealed runtimeclass MidiMessage32 : IMidiUniversalPacket,
```

Comment:
Each of those types also need to expose the additional WordN properties, so I don't think that will work correctly.

Reply:
That's okay. Multiple classes can share the same default interface. 

The default interface is just "the internal interface used at the ABI when the projection says the class name." It should be the interface that is most likely to be used by apps (since it is the most efficient to call methods on the default interface -- no QI needed).

For example, Windows.Media.MediaProperties.IMediaEncodingProperties is the default interface for AudioEncodingProperties, VideoEncodingProperties, ContainerEncodingProperties, ImageEncodingProperties, and TimedMediaEncodingProperties.

Reply:
> This also has the weird effect that using .as or .try_as in C++/WinRT will succeed in-between all of those classes (including the implementation classes - which can lead to hard to track type confusion bugs) since it checks for the default interface only - so `videoEncodingProperties.as<AudioEncodingProperties>()` will work. This is an easy to miss gotcha that has bitten me before.

I assume you mean when those classes all share the interface as their default interface?

So if MidiMessage32 and MidiMessage64 both have IMidiUniversalPacket as the default interface, then this would work?

`someMessage32.as<MidiMessage64>()`

If so, that would be unexpected (and unhelpful). I'll need to test that.

Reply:
Correct.

It will also allow the following:
```cpp
someMessage32.as<implementation::MidiMessage64>()
```
which can be very bad.